### PR TITLE
Feature: Add an unarchive query button

### PR DESCRIPTION
**Pull request by kyoshidajp**
Similar to #2180.

This is a query version.

# Screenshot

Confirm dialog:
unarchive_query_dialog

## Unarchive button

I can't make up my mind which one to choose where the button should be placed.

### Plan A(current version)

Same as #2180. Next to the query name.
unarchive_query_button

### Plan B

By the control button area.
unarchive_query_button2

### Plan C

Inside the botton group.
unarchive_query_botton3

### CODE REVIEW

**Comment by kocsmy** [comment]
Thanks for this @kyoshidajp —in order to keep this consistent with Unarchive Dashboard #2180 —I'd say Plan A makes the most sense.

**Comment by kocsmy** [comment]
Maybe in the future it'd make sense to have the label [Archived] with a dropdown to unarchive (so it's slightly easier to understand). Currently we don't really have a label+dropdown component, so the current solution will work for know.

**Comment by kyoshidajp** [comment]
@kocsmy Thanks for your comment.

@arikfr  Two issues are reported by codeclimate.  I think it's not a problem. WDYT?

**Comment by arikfr** [comment]
Actually I think we should go with Plan C for both queries and dashboards.

Re. CodeClimate: you can safely ignore those.

**Comment by kocsmy** [comment]
Oh, good point, we have ... menu there, too and that's where you archive already. Makes more sense.

**Comment by kyoshidajp** [comment]
OK! I'm going to change both queries and dashboards code tonight.

**Comment by kyoshidajp** [comment]
I've already changed.

However, I can't ignore CodeClimete error. "Disable check button" isn't shown.

codeclimate

File: client/app/pages/queries/view.js [hunk 1]
```diff
@@ -278,6 +279,26 @@ function QueryViewCtrl(
...
+        toastr.error('Query could not be unarchived.');
+      });
+    };
+
+    const title = 'Unarchive Query';
+    const message =
+      'Are you sure you want to unarchive this query?';
+    const confirm = { class: 'btn-warning', title: 'Unarchive' };
+
+    AlertDialog.open(title, message, confirm).then(unarchive);
```
Comment (arikfr) [inline client/app/pages/queries/view.js]: I'm not sure confirmation is needed here... it's not that destructive and easy to revert.

File: client/app/pages/queries/view.js [hunk 2]
```diff
@@ -278,6 +279,26 @@ function QueryViewCtrl(
     AlertDialog.open(title, message, confirm).then(archive);
   };
 
+  $scope.unarchiveQuery = () => {
+    const queryId = $scope.query.id;
+    const url = `api/queries/${queryId}/unarchive`;
+    const unarchive = () => {
+      Events.record('unarhive', 'queries', queryId);
+      $http.patch(url).success(() => {
```
Comment (arikfr) [inline client/app/pages/queries/view.js]: Why not use the `Query` resource?

File: redash/handlers/queries.py [hunk 3]
```diff
@@ -239,6 +239,17 @@ def delete(self, query_id):
         query.archive(self.current_user)
         models.db.session.commit()
 
+    def patch(self, query_id):
```
Comment (arikfr) [inline redash/handlers/queries.py]: Feels non idiomatic.. :\ Maybe a regular update (POST) request with `is_archived` set to `false`?

File: client/app/pages/queries/view.js (hunk 1 above)
Comment (kyoshidajp) [reply client/app/pages/queries/view.js]: I agree with you. I'll fix it.

**Comment by kyoshidajp** [comment]
I changed to Plan C and resolved a conflict.

unarchive_query_menu_button

Also, add a message.

unarchived_message

@arikfr Could you review this?

**Comment by gclsoft** [comment]
Why I still not find the unarchive query button button? @kyoshidajp 
image

**Comment by gclsoft** [comment]
@kyoshidajp I see, thank you.

**Comment by arikfr** [comment]
Hi,

(This is a template message, but I mean every word of it. Also you're welcome to reply)

Thank you for making this contribution. While we couldn't bring it to completion and merge, it's still very much appreciated. 🙇

In the past year the Redash code base gone under massive updates: on the backend we moved to Python 3 & RQ instead of Celery and on the frontend we replaced Angular with React. It's very likely this makes this PR irrelevant without significant changes. :-( 

I'm closing this PR now. But if you're still interested in making it happen, let me know and I will reopen.

Thanks.

**Pull request closed without merging**